tools/Evm/T8n/T8nCommand.cs

```diff
@@ -15,9 +29,67 @@ public static void Configure(ref CliRootCommand rootCmd)
         {
             var arguments = T8nCommandArguments.FromParseResult(parseResult);
 
-            T8nExecutor.Execute(arguments);
+            T8nOutput t8nOutput = new();
+            try
+            {
+                T8nExecutionResult t8nExecutionResult = T8nExecutor.Execute(arguments);
+
+                if (arguments.OutputAlloc == Stdout) t8nOutput.Alloc = t8nExecutionResult.Accounts;
+                else WriteToFile(arguments.OutputAlloc, arguments.OutputBaseDir, t8nExecutionResult.Accounts);
+
+                if (arguments.OutputResult == Stdout) t8nOutput.Result = t8nExecutionResult.PostState;
+                else WriteToFile(arguments.OutputResult, arguments.OutputBaseDir, t8nExecutionResult.PostState);
+
+                if (arguments.OutputBody == Stdout) t8nOutput.Body = t8nExecutionResult.TransactionsRlp;
+                else if (arguments.OutputBody is not null)
+                {
+                    WriteToFile(arguments.OutputBody, arguments.OutputBaseDir, t8nExecutionResult.TransactionsRlp);
+                }
+
+                if (t8nOutput.Body is not null || t8nOutput.Alloc is not null || t8nOutput.Result is not null)
+                {
+                    Console.WriteLine(_ethereumJsonSerializer.Serialize(t8nOutput, true));
+                }
+            }
+            catch (T8nException e)
+            {
+                t8nOutput = new T8nOutput(e.Message, e.ExitCode);
+            }
+            catch (IOException e)
+            {
+                t8nOutput = new T8nOutput(e.Message, T8nErrorCodes.ErrorIO);
+            }
+            catch (JsonException e)
+            {
+                t8nOutput = new T8nOutput(e.Message, T8nErrorCodes.ErrorJson);
+            }
+            catch (Exception e)
+            {
+                t8nOutput = new T8nOutput(e.Message, T8nErrorCodes.ErrorEvm);
+            }
+            finally
+            {
+                Environment.ExitCode = t8nOutput.ExitCode;
+                if (t8nOutput.ErrorMessage is not null)
+                {
+                    Console.WriteLine(t8nOutput.ErrorMessage);
+                }
+            }
         });
 
         rootCmd.Add(cmd);
     }
+
+    private static void WriteToFile(string filename, string? basedir, object outputObject)
+    {
+        if (basedir is not null)
+        {
+            basedir = basedir.TrimEnd('/');
+            basedir += '/';
+        }
```

Comment:
why needed?

Reply:
basedir can be with and without slash from user

Reply:
that shouldn't matter to the rest of the code though, ah I see the problem:

instead `FileInfo fileInfo = new(basedir + filename);` do `FileInfo fileInfo = new(Path.Combine(basedir, filename));` and you don't need to worry about separators